tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileRestorableReader.java

```diff
@@ -61,11 +61,12 @@ private void checkAndRepair() throws IOException {
     if (!isComplete()) {
       // Try to close it
       logger.info("File {} has no correct tail magic, try to repair...", file);
-      RestorableTsFileIOWriter rWriter =
-          new RestorableTsFileIOWriter(FSFactoryProducer.getFSFactory().getFile(file));
-      TsFileWriter writer = new TsFileWriter(rWriter);
-      // This writes the right magic string
-      writer.close();
+      try (RestorableTsFileIOWriter rWriter =
+          new RestorableTsFileIOWriter(FSFactoryProducer.getFSFactory().getFile(file))) {
+        TsFileWriter writer = new TsFileWriter(rWriter);
+        // This writes the right magic string
+        writer.close();
```

Comment:
remove this line?

Reply:
writer will not close automatically

Reply:
Who not remove `TsFileWriter writer = new TsFileWriter(rWriter);` into `try-with-resource` ?

Reply:
ok